src/www/ui/async/AjaxShowJobs.php

```diff
@@ -274,40 +274,42 @@ public function getShowJobsForEachJob($jobData, $forApi = false)
         $jobArr['jobQueue'][$key]['isReady'] = ($singleJobQueue['jq_end_bits'] ==
           1);
 
-        switch ($singleJobQueue['jq_type']) {
-          case 'readmeoss':
-            $jobArr['jobQueue'][$key]['download'] = "ReadMeOss";
-            break;
-          case 'spdx2':
-            $jobArr['jobQueue'][$key]['download'] = "SPDX2 report";
-            break;
-          case 'spdx2tv':
-            $jobArr['jobQueue'][$key]['download'] = "SPDX2 tag/value report";
-            break;
-          case 'spdx2csv':
-            $jobArr['jobQueue'][$key]['download'] = "SPDX2 CSV report";
-            break;
-          case 'dep5':
-            $jobArr['jobQueue'][$key]['download'] = "DEP5 copyright file";
-            break;
-          case 'reportImport':
-            $jobArr['jobQueue'][$key]['download'] = "uploaded SPDX2 report";
-            break;
-          case 'unifiedreport':
-            $jobArr['jobQueue'][$key]['download'] = "Unified Report";
-            break;
-          case 'clixml':
-            $jobArr['jobQueue'][$key]['download'] = "Clixml Report";
-            break;
-          case 'cyclonedx':
-            $jobArr['jobQueue'][$key]['download'] = "CycloneDX json Report";
-            break;
-          case 'decisionexporter':
-            $jobArr['jobQueue'][$key]['download'] = "FOSSology Decisions";
-            break;
-          default:
-            $jobArr['jobQueue'][$key]['download'] = "";
+        $reportName = "";
+        if ($this->showJobsDao->isJobIdPresentInReportGen($singleJobQueue['jq_job_fk'])) {
```

Comment:
Can you please explain what is the reason for this new check?

Reply:
Okay, so you want to hide the download links for old reports, if I understood it correctly.

Reply:
Yes